### PR TITLE
Provide a manual for the update-entry.py tool and an example commit for renewal

### PR DESCRIPTION
**Pull request by horazont**
This adds a README.rst to the data/ directory which shows a nice manual on how to manage your softwares entry.

Also an example commit where I renew aioxmpp with a commit message showing the renewal process in detail. I hope this is okay.

This doesn’t change any code which is run by the server or any content shown on the website (well, except that it renews aioxmpp).

### CODE REVIEW

File: data/README.rst [hunk 1]
```diff
@@ -0,0 +1,107 @@
+Software Directory README
+#########################
+
+As decided by the XSF Board, the Software Directory will need yearly renewal for software to not disappear from the listing on the website. To help software authors with this, scripts are provided which help you with creating a proper pull request.
```
Comment (guusdk) [inline data/README.rst]: I would remove "As decided by the XSF Board." It does not really matter who decided this for the README.

Comment (guusdk) [inline data/README.rst]: The text could mention more explicitly that you don't _have_ to use the tool, but can manually edit the JSON files too. Although using the tool will safeguard against formatting errors, a simple edit of the JSON files will be preferred by some, as you can do that without executing anything.

File: data/README.rst [hunk 2]
```diff
@@ -0,0 +1,107 @@
+Software Directory README
```
Comment (guusdk) [inline data/README.rst]: The ``rst`` extension of the README file is pretty uncommon. If we use a more common extension, more clients will automagically render the content properly. I'd suggest using no extension (just ``README``) or the markdown extension (``README.md``)

Comment (horazont) [inline data/README.rst]: Can you clarify? It renders fine on GitHub, and that’s handled server-side.

**Comment by guusdk** [comment]
I've added some minor suggestions - non of which are blocking issues for me.

File: data/README.rst (hunk 2 above)
Comment (guusdk) [inline data/README.rst]: Clarification: this is the very first time I've seen that extension being used. :)

File: data/README.rst (hunk 1 above)
Comment (horazont) [inline data/README.rst]: Thanks for your feedback, will integrate that into the PR today.

File: data/README.rst (hunk 2 above)
Comment (horazont) [inline data/README.rst]: Do you have concerns except the readability on GitHub and in plaintext (both of which is fine with rst in my opinion)?

Comment (guusdk) [inline data/README.rst]: I'm browsing the website source using various tools (pretty much whatever I have handy at the time.) Although I'm not sure if those tools will bork over the extension, why run the risk? Using a commonly used extension will prevent an issue, if there's one. 

If you do not agree, I wil happily shrug and move on. To me, it's a trivial issue at best. :)

**Comment by horazont** [comment]
@guusdk Reworded, mind to read over it again?

**Comment by guusdk** [comment]
I think it's a good improvement. I've already approved the PR, but will wait with the merge for a bit to give others the chance to provide feedback.

**Pull request merged**